### PR TITLE
fixed peer test

### PR DESCRIPTION
**Pull request by gluk256**
with a bit of refactoring

### CODE REVIEW

File: libdevcore/Log.h [hunk 1]
```diff
@@ -59,6 +59,14 @@ extern std::function<void(std::string const&, char const*)> g_logPost;
 /// or equal to the currently output verbosity (g_logVerbosity).
 extern std::map<std::type_info const*, bool> g_logOverride;
 
+/// Temporary changes system's verbosity for specific function. Restores the old verbosity when function returns.
+struct VerbosityHolder
+{
+	VerbosityHolder(int _temporaryValue) : oldLogVerbosity(g_logVerbosity) { g_logVerbosity = _temporaryValue; }
```
Comment (subtly) [inline libdevcore/Log.h]: space before colon

**Comment by gluk256** [comment]
fixed

**Comment by gavofyork** [comment]
fails on linux.

File: libdevcore/Log.h [hunk 2]
```diff
@@ -59,6 +59,14 @@ extern std::function<void(std::string const&, char const*)> g_logPost;
 /// or equal to the currently output verbosity (g_logVerbosity).
 extern std::map<std::type_info const*, bool> g_logOverride;
 
+/// Temporary changes system's verbosity for specific function. Restores the old verbosity when function returns.
+struct VerbosityHolder
```
Comment (arkpar) [inline libdevcore/Log.h]: This is not thread safe for general use. Please add a comment at least

**Comment by gluk256** [comment]
The warnings have been fixed.
Concerning the VerbosityHolder -- the corresponding comment is added. However, I think this comment was redundant, since I have only refactored the existing code, and so the VerbosityHolder is going to be used exactly the same as g_logVerbosity variable was used before.

**Pull request merged**